### PR TITLE
Add a healthcheck endpoint to the devserver

### PR DESCRIPTION
**Pull request by yamadapc**
This assists in implementing a few self-healing mechanisms.

### CODE REVIEW

File: packages/reporters/dev-server/src/Server.js [hunk 1]
```diff
@@ -495,6 +495,17 @@ export default class Server {
       setHeaders(res);
       next();
     });
+
+    app.use((req, res, next) => {
+      if (req.url === '/healthcheck') {
```
Comment (devongovett) [inline packages/reporters/dev-server/src/Server.js]: Could conflict with a user's URL (eg static page or API proxy), no? Maybe we should prefix with `__parcel_` like we do for the HMR endpoint?

**Review by devongovett — COMMENTED** [review]
Curious what you're planning on using this for

File: packages/reporters/dev-server/src/Server.js (hunk 1 above)
Comment (yamadapc) [reply packages/reporters/dev-server/src/Server.js]: Let me do that :+1:

**Comment by yamadapc** [comment]
@devongovett We sometimes have 'connectivity issues' impact the dev-loop due to broken tunnels; we want the dev-server for our main application to health-check and recover from that.

File: packages/reporters/dev-server/src/Server.js [hunk 2]
```diff
@@ -495,6 +495,17 @@ export default class Server {
       setHeaders(res);
       next();
     });
+
+    app.use((req, res, next) => {
+      if (req.url === '/__parcel_healthcheck') {
```
Comment (Nikola-3) [inline packages/reporters/dev-server/src/Server.js:500]: Is the `__` in `__parcel_healthcheck` meant to make it "private"?

**Review by Nikola-3 — APPROVED** [review]

File: packages/reporters/dev-server/src/Server.js (hunk 2 above)
Comment (yamadapc) [reply packages/reporters/dev-server/src/Server.js:500]: Yes, we're just trying to prevent clashes and be consistent with `__parcel_hmr`

**Pull request merged**